project-icp/bee-pollinator-app
# Frontend fetches real data

## Overview

The fetch endpoint for data already existed for a location/forage range/indicators. The front end has enough structure to request real data now instead of using mock dummy data. This PR hooks up the front and back ends officially and sets up the flow for when the database exists, authentication exists, and we save apiaries there.

I sketched this out as my train of thought. I can explain verbally if this doesn't make sense:
![img_20181116_105318](https://user-images.githubusercontent.com/10568752/48653519-dbcfcc80-e9d3-11e8-85b3-23f3a37277b3.jpg)

Connects #304 

### Demo

![back-front-connect](https://user-images.githubusercontent.com/10568752/48653802-ba6fe000-e9d5-11e8-92bf-780db5a528d3.gif)


![sort](https://user-images.githubusercontent.com/10568752/48653792-9f9d6b80-e9d5-11e8-8bf8-200ce89ea00a.gif)

### Notes

- I based this branch off the Geocoder branch accidentally. I'll fix later or hopefully the other PR gets approved and merged first haha

- If you click outside PA, the requests loop infinitely because the front end doesn't know how to handle server errors from the fetch endpoint. I'll make a followup issue.

## Testing Instructions

Click points within PA and try to use the dropdowns. All functionality should be maintained, plus real data now.
